runtime/taskfirecrackeropts.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"time"
+
+	"github.com/firecracker-microvm/firecracker-containerd/proto"
+	firecracker "github.com/firecracker-microvm/firecracker-go-sdk"
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+	"github.com/pkg/errors"
+)
+
+// overrideVMConfigFromTaskOpts overrides the firecracker VM config for the
+// task with values set in the protobuf message.
+func overrideVMConfigFromTaskOpts(
```

Comment:
I'm not sure if it's within scope of this PR, but I would rather throw away json configuration with giant override, and leave just protobuf stuff. So instead of `read json + build FC config + override protobuf` we would have `defaults + protobuf = FC Config`

Reply:
That's an interesting take. I'll open an issue and we can discuss that there?

Reply:
Yep :+1:

Reply:
@mxpv I took another look at the code.The JSON parsing that's happening seems to be build the default config. So, I'm not sure if that's relevant to this PR. IIUC, that's more of a question for creation of `service.config` is initialized. Was that your concern with the initial comment?

Reply:
Originally JSON was added as temp solution to simplify running. Here we have to manage two entities for configuration and keep `overrideVMConfigFromTaskOpts` to overwrite config params. This makes it more complicated than it should be and I see no reasons to keep old config. 
Anyway it's not a blocker for this PR, just concern :)

Reply:
Thanks for clarifying. Created #111 to keep track of that.